### PR TITLE
KAFKA-17559: Fix the flaky RemoteLogOffsetReader#testTaskQueueFullAndCancelTask

### PR DESCRIPTION
**Pull request by kamalcph**
### Committer Checklist (excluded from commit message)
- [ ] Verify design and implementation 
- [ ] Verify test coverage and CI build status
- [ ] Verify documentation (including upgrade notes)

### CODE REVIEW

File: core/src/test/java/kafka/log/remote/RemoteLogOffsetReaderTest.java
Comment (showuon) [inline core/src/test/java/kafka/log/remote/RemoteLogOffsetReaderTest.java]: OK, we now want to explicitly wait for all tasks completed because on CI, the machine is quite slow. Besides, I think we can also reduce the queue size to a smaller number, ex: 3, to use less resource to make the test more reliable.

Comment (kamalcph) [reply core/src/test/java/kafka/log/remote/RemoteLogOffsetReaderTest.java]: Reduce the queue size to 3.

**Review by mumrah — APPROVED** [review]
LGTM from a build perspective. I'll let others comment on the nature of the fix.

I just re-ran the tests for a third time. If this run passes, let's merge it and monitor this test on trunk (assuming another LGTM)

File: core/src/test/java/kafka/log/remote/RemoteLogOffsetReaderTest.java
Comment (satishd) [inline core/src/test/java/kafka/log/remote/RemoteLogOffsetReaderTest.java]: is there any reason to change it to 3? We are only submitting one async task, it needs to be at least 1.

**Review by satishd — COMMENTED** [review]
Thanks @kamalcph for the PR, left a minor comment.

**Review by satishd — APPROVED** [review]

**Review by showuon — APPROVED** [review]

**Pull request merged**
